numalogic/tools/callbacks.py

```diff
@@ -40,12 +39,14 @@ def disable(self):
 
     def on_train_epoch_end(self, trainer: pl.Trainer, pl_module: pl.LightningModule) -> None:
         super().on_train_epoch_end(trainer, pl_module)
-        metrics = self.get_metrics(trainer, pl_module)
+        loss = pl_module.total_train_loss / trainer.num_training_batches
```

Comment:
Is it guaranteed that `total_train_loss` is the result of `num_training_batches` additions?

Reply:
I didn't get your question here. `total_train_loss` is something we have defined, to capture the total loss in each mini-batch backpropagation. So mean loss across all batches can be gathered by dividing by `num_training_batches`.